tests/tests/publisher.rs

```diff
@@ -98,6 +142,57 @@ async fn publishes_transaction_for_each_published_block() {
         .is_ok_and(|result| result.is_some()));
 }
 
+// TODO: Make test more randomized for receipts
```

Comment:
This test doesn't cover all subject variants for `Receipts` and will be addressed in my follow-up PR